src/compiler/transformers/classFields.ts

```diff
@@ -193,20 +258,85 @@ namespace ts {
             return node;
         }
 
-        function visitPropertyDeclaration(node: PropertyDeclaration) {
+        function visitMethodOrAccessorDeclaration(node: MethodDeclaration | AccessorDeclaration) {
             Debug.assert(!some(node.decorators));
-            if (!shouldTransformPrivateFields && isPrivateIdentifier(node.name)) {
-                // Initializer is elided as the field is initialized in transformConstructor.
-                return factory.updatePropertyDeclaration(
-                    node,
-                    /*decorators*/ undefined,
-                    visitNodes(node.modifiers, visitor, isModifier),
-                    node.name,
-                    /*questionOrExclamationToken*/ undefined,
-                    /*type*/ undefined,
-                    /*initializer*/ undefined
+
+            if (!shouldTransformPrivateElements || !isPrivateIdentifier(node.name)) {
+                return visitEachChild(node, classElementVisitor, context);
+            }
+
+            // leave invalid code untransformed
+            const info = accessPrivateIdentifier(node.name);
+            Debug.assert(info, "Undeclared private name for property declaration.");

```

Comment:
Can a user write source text that gets into this case?

Reply:
I don't think so. That would indicate a bug in `addPrivateIdentifierToEnvironment`.